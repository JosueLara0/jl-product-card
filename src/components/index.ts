//? first pattern

//* components
import { ProductCard as ProductCardHOC } from "./ProductCard";
import { ProductButtons } from "./ProductButtons";
import { ProductImage } from "./ProductImage";
import { ProductTitle } from "./ProductTitle";
//* interfaces
import { ProductCardHOCProps } from "../interfaces/interfaces";

export const ProductCard: ProductCardHOCProps = Object.assign(ProductCardHOC, {
  Title: ProductTitle,
  Image: ProductImage,
  Buttons: ProductButtons,
});

//? second pattern

//* components
export * from "./ProductCard";
export { ProductButtons } from "./ProductButtons";
export { ProductTitle } from "./ProductTitle";
export { ProductImage } from "./ProductImage";
