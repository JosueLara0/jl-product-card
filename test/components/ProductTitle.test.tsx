//* libraries
import React from 'react';
import renderer from 'react-test-renderer';
//* components
import { ProductTitle, ProductCard } from '../../src/components';
//* data
import { product1 } from '../data/products';

describe('Test in ProductTitle', () => {
  test('should render component with the custom title', () => {
    const wrapper = renderer.create(<ProductTitle title="Custom Product" />);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('should render component with the product name', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>{() => <ProductTitle />}</ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
