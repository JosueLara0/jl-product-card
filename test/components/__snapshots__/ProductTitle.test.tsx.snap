// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in ProductTitle should render component with the custom title 1`] = `
<span
  className="productDescription undefined"
>
  Custom Product
</span>
`;

exports[`Test in ProductTitle should render component with the product name 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - Card without image
  </span>
</div>
`;
